core/domain/stats_domain.py

```diff
@@ -560,23 +587,31 @@ def __init__(self, calculation_output_type):
 
 class AnswerFrequencyList(AnswerCalculationOutput):
     """Domain object that represents a list of AnswerOccurrences."""
-    def __init__(self, answer_occurrences=None):
+    def __init__(self, answer_occurrences=None, limit=None):
         """Initialize domain object for answer frequency list for a given list
         of AnswerOccurrence objects (default is empty list).
         """
         super(AnswerFrequencyList, self).__init__(
             CALC_OUTPUT_TYPE_ANSWER_FREQUENCY_LIST)
-        self.answer_occurrences = (
-            answer_occurrences if answer_occurrences else [])
+        answer_occurrences = answer_occurrences or []
+        self._answer_counter = utils.OrderedCounter(
+            answer_occurrence.to_raw_pair()
+            for answer_occurrence in answer_occurrences)
+        self.limit = limit
+
+    def add_answer(self, answer):
+        """Adds a new Answer object."""
+        self._answer_counter[answer] += 1
 
-    def add_answer(self, answer_occurrence):
-        """Adds a new AnswerOccurrence object."""
-        self.answer_occurrences.append(answer_occurrence)
+    def add_answers(self, answers):
+        """Adds an iterable of new Answer objects."""
+        self._answer_counter.update(answers)
 
     def to_raw_type(self):
         return [
-            answer_occurrence.to_raw_type()
-            for answer_occurrence in self.answer_occurrences]
+            AnswerOccurrence(answer, frequency).to_raw_type()
+            for answer, frequency in (
+                self._answer_counter.most_common(self.limit))]
```

Comment:
Shouldn't have a limit here. The output of this method should fully represent the domain object.

Reply:
Done.